web/src/beta/features/Editor/tabs/story/SidePanel/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { FC, useMemo } from "react";
+
+import SidePanelCommon from "@reearth/beta/features/Editor/SidePanel";
+import { useT } from "@reearth/services/i18n";
+import { styled } from "@reearth/services/theme";
+
+import Action from "./Action";
+import Item from "./Item";
+import PageItemWrapper from "./PageItemWrapper";
+
+// TODO: these are currently rough definition
+type Props = {
+  stories: any;
+  selectedStory: any;
+  onSelectStory: (id: string) => void;
+  onStoryAdd: () => void;
+  selectedPageId?: string;
+  onSelectPage: (id: string) => void;
+  onPageAdd: () => void;
+};
+
+const SidePanel: FC<Props> = ({ onStoryAdd, onSelectStory, onPageAdd, onSelectPage }) => {
+  const t = useT();
+
+  const StoryContent = useMemo(() => {
```

Comment:
Can you extract the StoryContent and PageContent components to separate functional component?